jwst/assign_wcs/niriss.py

```diff
@@ -135,29 +135,25 @@ def niriss_soss(input_model, reference_files):
     try:
         # We'd like to open this file as a DataModel, so we can consolidate
         # the S3 URI handling to one place.  The S3-related code here can
-        # be removed once that has been changed.
+        # be removed once we have a specwcs DataModel subclass.
         if s3_utils.is_s3_uri(reference_files['specwcs']):
-            wl = asdf.open(s3_utils.get_object(reference_files['specwcs']))
+            uri = s3_utils.get_object(reference_files['specwcs'])
         else:
-            wl = asdf.open(reference_files['specwcs'])
-        with wl:
-            wl1 = wl.tree[1].copy()
-            wl2 = wl.tree[2].copy()
-            wl3 = wl.tree[3].copy()
-    except Exception:
-        raise IOError('Error reading wavelength correction from {}'.format(reference_files['specwcs']))
-
-    try:
-        velosys = input_model.meta.wcsinfo.velosys
-    except AttributeError:
-        pass
```

Comment:
No need for a try/except block here as it will never fail to have this attribute since it is in the wcsinfo schema.